libs/user/src/lib/auth/pages/profile-view/profile-view.component.ts

```diff
@@ -51,6 +54,20 @@ export class ProfileViewComponent implements OnInit {
     this.user$ = this.authQuery.user$;
     this.organization$ = this.organizationQuery.selectActive();
     this.previousPage = this.tunnelService.previousUrl || '../../..';
+
+    this.organization$.subscribe(org => {
+      if (!this.navLinks.some(link => link.path === 'preferences')) {
+        const app = getCurrentApp(this.routerQuery);
+        const appsWithPreferences: App[] = ['catalog', 'festival'];
+  
+        if (appsWithPreferences.includes(app) && org.appAccess[app].marketplace && !org.appAccess[app].dashboard) {
+          this.navLinks.push({
+            path: 'preferences',
+            label: 'Buyer Preferences'
+          })
+        }
+      }
+    })
   }
```

Comment:
```suggestion
    const hasPreferences = this.navLinks.some(link => link.path === 'preferences');
    if (hasPreferences) return;
    const org = this.organizationQuery.getActive();
    const app = getCurrentApp(this.routerQuery);
    if (app !== 'catalog' && app !== 'festival') return;
    const { marketplace, dashboard } = org.appAccess[app];
    if (marketplace && !dashboard) {
      this.navLinks.push({ path: 'preferences', label: 'Buyer Preferences' })
    }
  }
```

Would be a good idea to extract the logic of `canHavePreferences` : 
```typescript
function canHavePreferences(org: Organization, app: App) {
    if (app !== 'catalog' && app !== 'festival') return;
    const { marketplace, dashboard } = org.appAccess[app];
    return marketplace && !dashboard;
}
```